src/parser.zig

```diff
@@ -130,6 +130,8 @@ const Parser = struct {
             return dtb.Prop{ .Pinctrl2 = try self.integerList(u32, value) };
         } else if (std.mem.eql(u8, name, "assigned-clock-rates")) {
             return dtb.Prop{ .AssignedClockRates = try self.integerList(u32, value) };
+        } else if (std.mem.eql(u8, name, "device_type")) {
+            return dtb.Prop{ .DeviceType = value[0..value.len - 1] };
```

Comment:
Not sure if this is the best way to take the null-terminated string and turn it into a Zig slice. In my testing all the `device_type` values were null-terminated strings

Reply:
Indeed, it looks like they should be; see [`<string>`](https://devicetree-specification.readthedocs.io/en/stable/devicetree-basics.html#property-values-table) here. I'll add a test and refactor this out into a method for clarity, but yes, LGTM!